.github/workflows/macos-build.yml

```diff
@@ -0,0 +1,35 @@
+name: Build FStar Binaries
+
+on: [push, pull_request]
```

Comment:
can this be made nightly or weekly?

Reply:
We can also do this on a [schedule](https://help.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule). So if we say we want this weekly this would change to
```
on:
  schedule:
    - cron: '0 0 * * 0'
``` 
or any other cron schedule we want.